### ISSUE TITLE
Choosing initial temporary file

### ISSUE THREAD

**Issue by AC130USpectre** [spyder-ide/spyder#3495]
## Description
When I launch Spyder, it automatically open a temporary file in C:\Users\Username\.spyder-py3 directory. However, I want that spyder open a temporary file in my location (or, maybe, do not open any temporary file). I cannot choose properly the initial location for this file. I already choose a global working catalog in Preferences, and also tryied to edit a spyder.ini file (just remove any mention of .spyder directory). But Spyder continues to use incorrect initial location!
**What steps will reproduce the problem?**

1. Just everytime I launch Spyder I see this temporary file in incorrect location.

**What is the expected output? What do you see instead?**
I want to see that Spyder opens temporary file in the directory that I already set up before.

**Please provide any additional information below**

## Version and main components

* Spyder Version: 3.0.0 
* Python Version: 3.5.2
* Qt Versions: 5.6.0, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.3.0 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
pandas >=0.13.1  :  0.18.1 (OK)
numpy >=1.7      :  1.11.1 (OK)
sphinx >=0.6.6   :  1.4.6 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
sympy >=0.7.3    :  1.0 (OK)
pylint >=0.25    :  1.5.4 (OK)

```

**Comment by ccordoba12** [comment]
It is not possible to decide the location of that temporary file, sorry. And I don't think it's worth the effort to provide users with the ability to do that.

Maybe we could just remove that file. @jitseniesen, @goanpeca, @Nodd, what do you say?

**Comment by goanpeca** [comment]
I dont mind it. I actually use it for quick test scrpts or transient stuff

**Comment by jitseniesen** [comment]
> Maybe we could just remove that file.

I'm not sure what you mean, so let me try to spell out the details.

At the moment, if you open Spyder it automatically creates a tab in the editor connected to `.../.spyder/temp.py`. If you then open a new tab, it connects to `untitled0.py` in the global working directory (or another number). When you close all tabs, a new tab automatically appears, again connected to `untitled0.py` in the global working directory.

I would prefer it if the tab which is automatically created when you open Spyder uses the same convention, that is connecting to the global working directory. I think this does not prevent users from using it for quick test scripts or transient stuff.

Additionally, I would prefer it if Spyder did not automatically open a new tab after I close all editor tabs.

**Comment by ccordoba12** [comment]
> I dont mind it. I actually use it for quick test scrpts or transient stuff

This is the argument to have `temp.py`, i.e. that users can start doing some simple stuff without even saving a file.

> I would prefer it if the tab which is automatically created when you open Spyder uses the same convention, that is connecting to the global working directory

@jitseniesen, so you're in favor of removing `temp.py`?

>  I would prefer it if Spyder did not automatically open a new tab after I close all editor tabs.

This was the old behavior but it was an usability flaw in my opinion, and that's why I replaced it with the current one. In the old behavior, if you closed all tabs, then the Editor became a blank widget, devoid of any content, and that was ugly and confusing at the same time.

So that's why it won't come back :-)

**Comment by jitseniesen** [comment]
> @jitseniesen, so you're in favor of removing temp.py?

Yes.

A possible alternative for "quick test scripts and transient stuff": we can hide that there is a file in `.spyder` behind the tab to emphasize it is transient: Disable `Save`, and if the user does `Save As`, start the file name selector widget in the global working directory.

**Comment by fgotzens** [comment]
Just an idea: Why not having a look what other pieces of (modern) software do?
For example, most of the programs from Microsoft let the user choose what to do directly after opening (Open a new file, open an existing file, open an existing project, ...) 
A simple window would do the job, saves us time and gets us rid of the temp.py file ...

**Comment by ccordoba12** [comment]
Yeah, we're thinking to do something like that for Spyder 4.

**Comment by MNRK01** [comment]
It's good to see additional choices for the "start-up file" in version 4.0. Will one of the choices be to not have any file upon start-up?

I predominantly use Spyder's IPython console and simply close the editor pane with Ctrl+Shift+E every time I start-up Spyder. So from my perspective, it would be good to let the user choose their start-up interface, even one without an Editor pane.

Thanks for the excellent work.

**Comment by goanpeca** [comment]
Is the idea here to:

1.) Add a widow to select what the user can do?
2.) Add these options as the content of the `black editor widget`?

**Comment by MNRK01** [comment]
Assuming that @goanpeca  is asking his questions for me.

If it were me, I would add an option under ```Tools > Preferences > General > Advanced Settings``` that allows a user to choose if they don't want an editor window at all (i.e. not even a blank editor widget). Maybe the default is the current blank editor widget display so that this choice does not disrupt the expectations of most users.

I ask for this feature since today (as an example) one can go to ```View > Panes``` and select/deselect a pane (e.g. File Explorer or Outline). When Spyder restarts, that choice is respected and the users chosen widget arrangement is retained. However, in this respect the Editor is different - it redisplays no matter that a user may have chosen to not have it displayed prior to restarting Spyder. All I am asking is to give the user the same choice for the Editor pane as other widgets.

Thanks!

**Comment by ccordoba12** [comment]
> However, in this respect the Editor is different - it redisplays no matter that a user may have chosen to not have it displayed prior to restarting Spyder

I agree we're giving the Editor special treatment and that that should not be the case. @MNRK01, that's not a hard fix, in case you want to try it (we did something similar for the IPython console).

**Comment by MNRK01** [comment]
OK, I will dig into the code and try to propose that change. I will find and follow your IPython console fix as you recommend.

**Comment by ccordoba12** [comment]
This is what we did in the IPython console:

https://github.com/spyder-ide/spyder/blob/0397e98adc903b95ddbaee669e2d4f448c1d349d/spyder/plugins/ipythonconsole/plugin.py#L221

`toggle_view` controls what to do when a plugin's entry in the menu `View > Panes` is activate or not.
